src/main/java/net/sf/jabref/gui/SidePaneManager.java

```diff
@@ -75,6 +79,21 @@ public synchronized void toggle(String name) {
         }
     }
 
+    /**
+     * if panel is hidden it will be shown and focused
+     * if panel is visible but not focused it will be focused
+     * ig panel is visible and focused it will be hidden
+     *
+     * @param name name of the panel
+     */
+    public synchronized void toggleThreeWay(String name){
+        if (isComponentVisible(name) && Globals.getFocusListener().getFocused() == components.get(name)) {
```

Comment:
Would be nice if you could add additional brackets here, that would make it clearer.